pkg/destroy/gcp/filestore.go

```diff
@@ -0,0 +1,92 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/api/file/v1"
+
+	gcpconsts "github.com/openshift/installer/pkg/constants/gcp"
+)
+
+func (o *ClusterUninstaller) filestoreParentPath() string {
+	return fmt.Sprintf("projects/%s/locations/-", o.ProjectID)
+}
+
+func (o *ClusterUninstaller) clusterFilestoreLabelFilter() string {
+	return fmt.Sprintf("labels.%s = \"owned\"", fmt.Sprintf(gcpconsts.ClusterIDLabelFmt, o.ClusterID))
+}
+
+func (o *ClusterUninstaller) listFilestores(ctx context.Context) ([]cloudResource, error) {
+	o.Logger.Debug("Listing filestores")
+	result := []cloudResource{}
+
+	ctx, cancel := context.WithTimeout(ctx, defaultTimeout)
+	defer cancel()
+	instSvc := file.NewProjectsLocationsInstancesService(o.fileSvc)
+	lCall := instSvc.List(o.filestoreParentPath()).Context(ctx).Filter(o.clusterFilestoreLabelFilter())
+
+	nextPageToken := "pageToken"
+	for nextPageToken != "" {
+		instances, err := lCall.Do()
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving filestore instances: %w", err)
+		}
+
+		for _, activeInstance := range instances.Instances {
+			if len(activeInstance.FileShares) == 0 {
+				// skip multi-share instances
+				o.Logger.Debugf("Skipping multi-share filestore %s", activeInstance.Name)
+				continue
+			}
+			o.Logger.Debugf("Found filestore %s", activeInstance.Name)
+			result = append(result, cloudResource{
+				name:     activeInstance.Name,
+				typeName: "filestore",
+			})
+		}
+
+		nextPageToken = instances.NextPageToken
+		lCall.PageToken(nextPageToken)
+	}
+	return result, nil
+}
+
+func (o *ClusterUninstaller) deleteFilestore(ctx context.Context, item cloudResource) error {
+	o.Logger.Debugf("Deleting filestore %s", item.name)
+
+	ctx, cancel := context.WithTimeout(ctx, defaultTimeout)
+	defer cancel()
+	instSvc := file.NewProjectsLocationsInstancesService(o.fileSvc)
+	_, err := instSvc.Delete(item.name).Context(ctx).Do()
+	if err != nil && !isNoOp(err) {
+		o.resetRequestID(item.typeName, item.name)
+		return fmt.Errorf("failed to delete filestore %s", item.name)
+	}
+	if err != nil && isNoOp(err) {
```

Comment:
Shouldn't this be
```suggestion
	if err == nil || isNoOp(err) {
```
so you also remove the filestore from the pending items in case of successful `Delete`?

Reply:
Yes, this looks weird at the first sight. I'm almost sure I copied that logic from some other place (disk.go maybe?). The `destroyFilestores` method gets called in a loop, which means the item would get removed from pending in the next attempt (essentially verifying the filestore really doesn't exist, since the `Delete` call should return 404).

I even wonder if this whole `if` block is needed: at this point it's clear the call either succeeded or the error is benign (`isNoOp`) so there's probably no point resetting the request and we can happily remove the item from pending and log success.